eos-tech-support/eos-user-stress-test

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SLEEP_APP_LAUNCH=30
+SLEEP_URL_LAUNCH=10
+
+command=""
+if [ $# -gt 0 ]; then
+        command=$1
+fi
+
+check_dep_flatpak() {
+    # relies on set -e
+    flatpak info $1 > /dev/null
+}
+
+check_deps_light() {
+    check_dep_flatpak "com.endlessm.animals.en"
+}
+
+check_deps_medium() {
+    # empty function bodies not allowed in strict mode
+    true
+}
+
+check_deps_heavy() {
+    check_dep_flatpak "org.gimp.Gimp"
+}
+
+check_deps() {
+    check_deps_light
+    check_deps_medium
+    check_deps_heavy
+}
+
+command_launch() {
+    cmd="$1"
+    shift
+
+    $cmd $@ &
+    # wait to simulate a (somewhat-patient) user launching an app then waiting
+    # some time before doing any other work (like launching other apps)
+    #
+    # NOTE: a more-realistic scenario would be waiting until the app appears to
+    # be responsive, use it for a while, then launch another app. But that would
+    # be fairly complex to simulate as an app's "apparent" readiness to be used
+    # depends on the app and is probably not readily detected in software.
+    sleep $SLEEP_APP_LAUNCH
+}
+
+url_launch() {
+    gio open $1 &
+
+    # see note above for command_launch; this wait time is shorter as users seem
+    # likely to wait much less time between opening new tabs
+    sleep_time=$SLEEP_URL_LAUNCH
+
+    if [ $browser_launched -eq 0 ]; then
+        # on the first URL launch of this script's run, assume the browser is
+        # launching and give it more time.
+        #
+        # this isn't a perfect check but the browser (or any apps but the
+        # terminal and maybe system monitor) shouldn't be running at the start
+        # of this script under normal use.
+        sleep_time=$SLEEP_APP_LAUNCH
+        browser_launched=1
+    fi
+
+    sleep $sleep_time
+}
+
+browser_launched=0
+
+# note that this implicitly depends upon EOS >= 3.0 for the `gio` command and
+# apps being packaged as Flatpaks
+case $command in
+    check)
+        check_deps
+        echo "all dependencies fulfilled"
+        ;;
+    light)
+        check_deps_light
+        command_launch flatpak run com.endlessm.animals.en
+        command_launch lowriter --norestore files/chicken.odt
+        url_launch https://www.google.com/search?q=chicken
+        url_launch https://www.youtube.com/watch?v=dfR_LdA3fPI
+        url_launch https://inbox.google.com
```

Comment:
In my use of the script, I stopped bothering to log in since it's a hassle. I think it makes sense in a realistic test case but I think one that is interactive. This test case is meant to be as automated as possible, so I've replaced this URL with an account-free page.